apps/aecore/src/aec_peer_analytics.erl

```diff
@@ -0,0 +1,141 @@
+log_peer_info(Pub, NetID, Ver, Os, Rev, Vendor) ->
+    gen_server:call(?MODULE, {log_peer_info, Pub, NetID, Ver, Os, Rev, Vendor}).
+log_peer_info_unknown(Pub) ->
+    gen_server:call(?MODULE, {log_peer_info_unknown, Pub}).
+get_stats() -> gen_server:call(?MODULE, get_stats).
+enabled() -> aeu_env:user_config([<<"sync">>, <<"peer_analytics">>], false).
+
+start_link() ->
+    gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
+
+-spec init(any()) -> {ok, state()}.
+init(_) ->
+    Enabled = enabled(),
+    case Enabled of
+        true ->
+            lager:info("Enabling detailed peer statistics"),
+            application:set_env(aecore, sync_max_outbound, 1000000),
+            application:set_env(aecore, sync_max_inbound, 1000000);
+        false -> lager:info("Detailed peer statistics are disabled")
+    end,
+    {ok, #{enabled => Enabled, stats => #{}, probes_pids => #{}, probes_mref => #{}}}.
+
+handle_call({log_peer_info, Pub, NetID, Ver, Os, Rev, Vendor}, _From, #{ stats := Stats } = St) ->
+    Info = #{ network_id => NetID
+            , version => Ver
+            , os => Os
+            , revision => Rev
+            , vendor => Vendor
+            },
+    {reply, ok, St#{stats => update_peer_info(Stats, Pub, Info)}};
+handle_call({log_peer_info_unknown, Pub}, _From, #{ stats := Stats } = St) ->
+    {reply, ok, St#{stats => update_peer_info(Stats, Pub, unknown)}};
+handle_call(get_stats, _From, #{stats := Stats } = St) -> {reply, Stats, St};
+handle_call(Msg, _From, #{} = St) ->
+    lager:debug("Unhandled call in peer analytics ~p", [Msg]),
+    {noreply, St}.
+
+handle_info({'DOWN', MRef, process, _, _}, #{probes_pids := PPids, probes_mref := PRefs} = St) ->
+    Pub = maps:get(MRef, PRefs),
+    {noreply, St#{probes_pids := maps:remove(Pub, PPids), probes_mref := maps:remove(MRef, PRefs)}}.
+
+handle_cast(_Msg, #{enabled := false} = St) -> {noreply, St};
+handle_cast({log_temporary_peer_status, #{r_pubkey := Pub, host := Host, port := Port}, GHash, THash, Diff}, St) ->
+    St1 = log_ping(St, Pub, Host, Port, GHash, THash, Diff),
+    {noreply, St1};
+handle_cast({log_peer_status, #{r_pubkey := Pub, host := Host, port := Port}, GHash, THash, Diff}, St) ->
+    St1 = log_ping(St, Pub, Host, Port, GHash, THash, Diff),
+    St2 = maybe_schedule_version_probe(St1, Pub),
+    {noreply, St2};
+handle_cast(Msg, #{} = St) ->
+    lager:debug("Unhandled cast in peer analytics ~p", [Msg]),
+    {noreply, St}.
+
+log_ping(#{stats := Stats} = St, Pub, Host, Port, GHash, THash, Diff) ->
+    PS0 = maps:get(Pub, Stats, #{ info => unresolved, first_seen => unix_time() }),
+    PS = PS0#{ host => Host
+             , port => Port
+             , last_seen => unix_time()
+             , genesis_hash => GHash
+             , top_hash => THash
+             , difficulty => Diff
+             },
+    St#{stats => Stats#{ Pub => PS }}.
+
+unix_time() ->
+    {MegaSecs, Secs, _MicroSecs} = now(),
+    MegaSecs * 1000000 + Secs.
```

Comment:
`erlang:system_time(second)` will give the same result, only more straightforwardly.
(Except `now()` is defined to always increase monotonically. If this is desired,
`erlang:monotonic_time(second) + erlang:time_offset(second)` should do essentially the same.)